pallets/ethereum-transactions/src/ethereum_transaction.rs

```diff
@@ -424,6 +426,10 @@ pub struct EthSignatures {
 }
 
 impl EthSignatures {
+    fn max_encoded_len() -> usize {
+        u32::MAX as usize
+    }
```

Comment:
I don't think this is correct. This is the definition of the struct
```
pub struct EthSignatures {
    pub signatures_list: Vec<ecdsa::Signature>,
}
```
and `u32::MAX as usize` doesn't seem plausible for a bound of the encoded data.

In general we shouldn't implement `max_encoded_len` unless we have to, usually if you use bounded data structures all you need is to derive `MaxEncodedLen` as it was done above.
I think we should make the `signatures_list` a bounded data structure itself instead.
The good news is that the list of signatures can use the same bound as the bound we have for validators.

Reply:
So, I was doing some checks on this, just if it helps. If we _absolutely want to compute_ this maxLen, we still can.
The BoundedVec is still SCALE-encoded as a dynamic vector, which means it will only list the actual elements of the vector, and will not create empty elements for those remaining slots the vector allows for.

But it also means that the first few bytes of the vector encoding specify how many elements actually are in the vector. And this length encoding has some specific size. [Details here](https://docs.substrate.io/reference/scale-codec/#fn-1).

If the vector's length is up to 63, the prefix will be 1-byte long. If  it is up to 16,383 it will be 2-bytes long. If it is up to `2^30 - 1`, it will be 4 bytes long. And beyond that, I don't think we need to worry.

So the actual MAX_LEN would be `2 + element_size * number_of_elements` if the bound is below 16,384, or `4 + element_size * number_of_elements` if it is above.

Reply:
Just noticed that we'll probably be below 64, in which case the prefix is just `1 byte`.